cmd/antrea-agent/util_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"antrea.io/antrea/pkg/agent/util"
+)
+
+func TestGetAvailableNodePortAddresses(t *testing.T) {
+	testCases := []struct {
+		name                        string
+		nodePortAddressesFromConfig []string
+		excludeDevices              []string
```

Comment:
Since this is not a factor that affects anything, no need to include it as a test factor.
https://github.com/antrea-io/antrea/pull/4349#discussion_r1032523487

Reply:
Yeah but to call the function we need `excludeDevices` (i.e getAvailableNodePortAddresses(tc.nodePortAddressesFromConfig, tc.excludeDevices) so I've added values for the excludeDevices in test cases

Reply:
`getAvailableNodePortAddresses` doesn't have to only use parameters from test cases.
it can just be `getAvailableNodePortAddresses(tc.nodePortAddressesFromConfig, []string{})`, passing `[]string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"}` is pointless unless you check the received parameter of `getAllNodeAddresses`

Reply:
Addressed